submissions/Siusarna/port-sniffer/sniffer.js

```diff
@@ -0,0 +1,139 @@
+
+               Example: node sniffer.js --host google.com --port 100-2345
+
+    --help     Parameter will show this instructions.
+`;
+
+const validHost = async host => {
+  if (!host) {
+    process.stdout.write(
+      'Invalid host parameter. Try node sniffer.js --help \n'
+    );
+  }
+  try {
+    return await dns
+      .lookup(host)
+      .then(res => {
+        return res.address;
+      })
+      .catch(error => {
+        process.stdout.write(
+          `Invalit host name. ${error}. Try node sniffer.js --help \n`
+        );
+        return 0;
+      });
+  } catch (e) {
+    process.stdout.write(e);
+    return 0;
+  }
+};
+
+const validatorForPort = port => {
+  let validPort = [1, 65535];
+  if (port) {
+    validPort = port.split('-').map(el => Number(el));
+    if (
+      validPort[0] < 0 ||
+      validPort[1] > 65535 ||
+      validPort[2] ||
+      validPort[0] > validPort[1]
+    ) {
+      throw new Error('Invalid port range. See help \n');
+    }
+  } else {
+    process.stdout.write(
+      'No port range specified. Setting to default. For more information try --help \n'
+    );
+  }
+  return validPort;
+};
+
+const checkConnection = async (port, host) => {
+  let flag = false;
+  await new Promise((resolve, reject) => {
+    const socket = new net.Socket();
+    socket.setTimeout(300);
+    socket.on('connect', () => {
+      process.stdout.write('.');
+      flag = true;
+      socket.destroy();
+      resolve();
+    });
+    socket.on('timeout', () => {
+      socket.destroy();
+      reject();
+    });
+    socket.on('error', () => {
+      socket.destroy();
+      reject();
+    });
+    socket.connect(port, host);
+  }).catch(e => e);
+  if (flag) {
+    return port;
+  }
+  return null;
+};
+
+const findOpenPorts = async (ports, host) => {
+  let openPorts = [];
+  for (let i = ports[0]; i <= ports[1]; i += 1) {
+    openPorts.push(checkConnection(i, host));
+  }
+  openPorts = await Promise.all(openPorts);
+  return openPorts.filter(Boolean);
+};
+
+const sniffer = async ({ host, port, helpFlag }) => {
+  if (helpFlag) {
+    return 0;
+  }
+  if (!host && !port) {
+    process.stdout.write('No arguments. Try node sniffer.js --help. \n');
+    return 0;
+  }
+  const checkPort = validatorForPort(port);
+  const checkHost = await validHost(host);
+  const result = await findOpenPorts(checkPort, checkHost);
+  process.stdout.write('\n');
+  for (let i = 0; i < result.length; i += 1) {
+    if (i !== result.length - 1) {
+      process.stdout.write(`${result[i].toString()},`);
+    } else {
+      process.stdout.write(result[i].toString());
+    }
+  }
+  process.stdout.write(' ports are opened \n');
+  return 0;
+};
+
+const parseArgs = () => {
+  const obj = {
+    help: false
+  };
+  if (args.indexOf('--help') !== -1) {
+    process.stdout.write(help);
+    obj.help = true;
+  }
+  const indexPort = args.indexOf('--ports');
```

Comment:
I suggest using `Array.prototype.find` or `Array.prototype.includes` instead of `indexOf` to check if an item is in array.

Reply:
Can I use `Array.prototype.findIndex` ?

Reply:
Both `Array.prototype.indexOf` and `Array.prototype.findIndex` basically say: "Find me the position of the element in the array". 
`Array.prototype.includes`  says: "I don't care where the element is, I just need to know if it is there." 

If I understand your intention correctly, you only need to know if the element is included in the array and you do not use the index itself. 
`Array.prototype.indexOf` or `Array.prototype.findIndex` both work fine, but they require slightly more code and they don't show your intention as clear as `Array.prototype.includes`.

Reply:
I understand, but`Array.prototype.indexOf` or `Array.prototype.findIndex`  gives me index of the key ("--host" for example). In order to take the argument I write `const host = args[Array.prototype.indexOf("--host")+1]

Reply:
Oh, I see. I have missed that. Then you are right, let's leave it as it is!